xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingMarketDataService.java

```diff
@@ -57,7 +58,7 @@ public KrakenStreamingMarketDataService(KrakenStreamingService service) {
                 .map(ob -> {
                     KrakenOrderBookStorage orderBook = ob.toKrakenOrderBook(orderBooks.get(channelName), depth);
                     orderBooks.put(channelName, orderBook);
-                    return KrakenAdapters.adaptOrderBook(orderBook.toKrakenDepth(), currencyPair);
+                    return KrakenOrderBookUtils.verifyKrakenOrderBook(KrakenAdapters.adaptOrderBook(orderBook.toKrakenDepth(), currencyPair));
```

Comment:
This all looks like it should occur inside `adaptOrderBook` BEFORE creation of the `OrderBook` object.

Reply:
I would have put it on adaptOrderbook but we have some version issues and that is why i thought it would be better to have it on bitrich-info project instead.